rustls-platform-verifier/src/verification/android.rs

```diff
@@ -77,21 +77,42 @@ impl Verifier {
         Self {
             #[cfg(any(test, feature = "ffi-testing"))]
             test_only_root_ca_override: None,
-            default_provider: OnceCell::new(),
+            crypto_provider: OnceCell::new(),
         }
     }
 
+    /// Attempts to set the [`CryptoProvider`] for this `Verifier`.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if this verifier's crypto provider has already been set.
+    pub fn set_provider(&mut self, crypto_provider: Arc<CryptoProvider>) {
+        self.crypto_provider
+            .set(crypto_provider)
+            .expect("crypto provider was already set");
+    }
+
+    /// Chainable constructor to set the [`CryptoProvider`] on a new [`Verifier`]
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if this verifier's crypto provider has already been set.
+    pub fn with_provider(mut self, crypto_provider: Arc<CryptoProvider>) -> Self {
```

Comment:
I wonder if it would be better/possible to raise the `set_provider` and `with_provider` methods (and their `impl` block) into the `verifier/mod.rs` level. This would help cut down on the code and comment duplication. As the platform-specific modules don't need to call these, we wouldn't end up with any circular code either.

As an example of what I mean:
```rust
<snip>

#[cfg(windows)]
mod windows;

#[cfg(windows)]
pub use windows::Verifier;

impl Verifier {
    pub fn set_provider(&mut self) { ... }
    pub fn with_provider(mut self) { ... }
}

<snip>
```

Reply:
Will do, I like that but figured there was a reason there wasn't already a shared-behavior block. We'll need to omit `feature = "other"` which doesn't have a crypto provider, but that isn't much complexity